packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -206,15 +213,57 @@ void diagnoseXcodeBuildFailure(XcodeBuildResult result) {
     printError("  open ios/Runner.xcodeproj");
     return;
   }
+  if (result.buildForDevice
+      && result.buildCommands != null
+      && result.buildCommands.contains('build')
+      && result.appDirectory != null) {
```

Comment:
When buildForDevice is true, are buildCommands and appDirectory always expected? If so, use assertions ahead of the conditional.